### ISSUE TITLE
shoutrrr slack notifications not working for API botuser and not working correctly for webhook

### ISSUE THREAD

**Issue by nunnsby** [containrrr/watchtower#1424]
### Describe the bug

2 issues when using docker-compose. I am not sure if this is specific to shoutrrr, or watchtower, so posting here.

1) The Bot Auth token is not sending at all, and produces a "Failed to send message" 
2) The webhook token does post, but does not post any of the override manual details.
---
### 1) Bot auth token config: (I have randomised my token and channel details)
```
      - WATCHTOWER_NOTIFICATIONS=shoutrrr
      - WATCHTOWER_NOTIFICATION_URL=slack://xoxb:123456789-123456789-mytokendetailsremoved@C0123CHANNEL?color=good&title=Great+News&icon=man-scientist&botname=Shoutrrrbot
```
Output of running watchtower with API botuser: (Failed to send shoutrrr notification error - line 5)
```
Attaching to watchtower
watchtower    | time="2022-10-11T21:37:12+01:00" level=info msg="Watchtower 1.4.0"
watchtower    | time="2022-10-11T21:37:12+01:00" level=info msg="Using notifications: slack"
watchtower    | time="2022-10-11T21:37:12+01:00" level=info msg="Only checking containers using enable label"
watchtower    | time="2022-10-11T21:37:12+01:00" level=info msg="Running a one time update."
watchtower    | time="2022-10-11T21:37:13+01:00" level=error msg="Failed to send shoutrrr notification" error="failed to send slack notification: api response: missing_scope" index=0 notify=no service=slack
watchtower    | time="2022-10-11T21:37:14+01:00" level=info msg="Session done" Failed=0 Scanned=2 Updated=0 notify=no
watchtower    | time="2022-10-11T21:37:14+01:00" level=info msg="Waiting for the notification goroutine to finish" notify=no
watchtower exited with code 0
```
---
### 2) Webhook token with manual details, outputs no errors, but Slack does not show any updated / manual details:
```
      - WATCHTOWER_NOTIFICATIONS=shoutrrr
      - WATCHTOWER_NOTIFICATION_URL=slack://hook:HOOK-DETAILS-HAVEBEENREMOVED@webhook?color=good&title=Great+News&icon=man-scientist&botname=Shoutrrrbot
```
Output of running watchtower with webhook:
```
Attaching to watchtower
watchtower    | time="2022-10-11T21:42:51+01:00" level=info msg="Watchtower 1.4.0"
watchtower    | time="2022-10-11T21:42:51+01:00" level=info msg="Using notifications: slack"
watchtower    | time="2022-10-11T21:42:51+01:00" level=info msg="Only checking containers using enable label"
watchtower    | time="2022-10-11T21:42:51+01:00" level=info msg="Running a one time update."
watchtower    | time="2022-10-11T21:42:52+01:00" level=info msg="Session done" Failed=0 Scanned=2 Updated=0 notify=no
watchtower    | time="2022-10-11T21:42:52+01:00" level=info msg="Waiting for the notification goroutine to finish" notify=no
watchtower exited with code 0
```
Screenshot from slack channel. Channel = web-services, app name as created in slack is temp-test. As you can see, the only override detail that carried over is the color (green / good). The title didn't change, and neither did the icon appear/change, nor the did botname override.

![2022-10-11 web-services](https://user-images.githubusercontent.com/9913478/195194779-47958bb9-8ee2-42b2-b6d2-bb1a2942affc.png)

### Steps to reproduce

See detail above in compose.

### Expected behavior

1) Bot user to send to channel.
2) webhook to change overrides.

### Environment

Docker version 20.10.18, build b40c2f6
Watchtower version: latest - 333de6ea525a
```
$ docker pull containrrr/watchtower
Using default tag: latest
latest: Pulling from containrrr/watchtower
Digest: sha256:bbf9794a691b59ed2ed3089fec53844f14ada249ee5e372ff0e595b73f4e9ab3
Status: Image is up to date for containrrr/watchtower:latest
docker.io/containrrr/watchtower:latest
```

### Your logs

```text
Logs included. They are the same whether trying to send webhook or bot user. There is no mention of shoutrrr in any logs.


Recreating watchtower ... done
Attaching to watchtower
watchtower    | time="2022-10-11T22:09:55+01:00" level=debug
watchtower    | time="2022-10-11T22:09:55+01:00" level=debug msg="Sleeping for a second to ensure the docker api client has been properly initialized."
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Making sure everything is sane before starting"
watchtower    | time="2022-10-11T22:09:56+01:00" level=info msg="Watchtower 1.4.0"
watchtower    | time="2022-10-11T22:09:56+01:00" level=info msg="Using notifications: slack"
watchtower    | time="2022-10-11T22:09:56+01:00" level=info msg="Only checking containers using enable label"
watchtower    | time="2022-10-11T22:09:56+01:00" level=info msg="Running a one time update."
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Checking containers for updated images"
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Retrieving running containers"
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Trying to load authentication credentials." container=/watchtower image="containrrr/watchtower:latest"
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="No credentials for containrrr found" config_file=/config.json
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Got image name: containrrr/watchtower:latest"
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Checking if pull is needed" container=/watchtower image="containrrr/watchtower:latest"
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Building challenge URL" URL="https://index.docker.io/v2/"
watchtower    | time="2022-10-11T22:09:56+01:00" level=error msg="Failed to send shoutrrr notification" error="failed to send slack notification: api response: missing_scope" index=0 notify=no service=slack
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Got response to challenge request" header="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\"" status="401 Unauthorized"
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Checking challenge header content" realm="https://auth.docker.io/token" service=registry.docker.io
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Setting scope for auth token" image=containrrr/watchtower scope="repository:containrrr/watchtower:pull"
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="No credentials found."
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Parsing image ref" host=index.docker.io image=containrrr/watchtower normalized="docker.io/containrrr/watchtower:latest" tag=latest
watchtower    | time="2022-10-11T22:09:56+01:00" level=debug msg="Doing a HEAD request to fetch a digest" url="https://index.docker.io/v2/containrrr/watchtower/manifests/latest"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Found a remote digest to compare with" remote="sha256:bbf9794a691b59ed2ed3089fec53844f14ada249ee5e372ff0e595b73f4e9ab3"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg=Comparing local="sha256:bbf9794a691b59ed2ed3089fec53844f14ada249ee5e372ff0e595b73f4e9ab3" remote="sha256:bbf9794a691b59ed2ed3089fec53844f14ada249ee5e372ff0e595b73f4e9ab3"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Found a match"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="No pull needed. Skipping image."
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="No new images found for /watchtower"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Trying to load authentication credentials." container=/https-portal image="steveltn/https-portal:latest"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="No credentials for steveltn found" config_file=/config.json
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Got image name: steveltn/https-portal:latest"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Checking if pull is needed" container=/https-portal image="steveltn/https-portal:latest"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Building challenge URL" URL="https://index.docker.io/v2/"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Got response to challenge request" header="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\"" status="401 Unauthorized"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Checking challenge header content" realm="https://auth.docker.io/token" service=registry.docker.io
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Setting scope for auth token" image=steveltn/https-portal scope="repository:steveltn/https-portal:pull"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="No credentials found."
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Parsing image ref" host=index.docker.io image=steveltn/https-portal normalized="docker.io/steveltn/https-portal:latest" tag=latest
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Doing a HEAD request to fetch a digest" url="https://index.docker.io/v2/steveltn/https-portal/manifests/latest"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Found a remote digest to compare with" remote="sha256:d0f9bb4fa9f2c9a1a4bb8b75a91897f95ca82e0c988f462f4172bf468c666e13"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg=Comparing local="sha256:d0f9bb4fa9f2c9a1a4bb8b75a91897f95ca82e0c988f462f4172bf468c666e13" remote="sha256:d0f9bb4fa9f2c9a1a4bb8b75a91897f95ca82e0c988f462f4172bf468c666e13"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="Found a match"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="No pull needed. Skipping image."
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="No new images found for /https-portal"
watchtower    | time="2022-10-11T22:09:57+01:00" level=debug msg="This is the watchtower container /watchtower"
watchtower    | time="2022-10-11T22:09:57+01:00" level=info msg="Session done" Failed=0 Scanned=2 Updated=0 notify=no
watchtower    | time="2022-10-11T22:09:57+01:00" level=info msg="Waiting for the notification goroutine to finish" notify=no
watchtower exited with code 0
```
```


### Additional context

_No response_

**Comment by piksel** [comment]
There are so many different things here.

The first one should work, but is probably a shoutrrr bug (`missing_scope` - perhaps the the slack API have changed?).

The second one, as it's not using the API, is limited to the webhook limitations. That means no override of name and/or icon.

The title is always overridden by watchtower, so it doesn't matter what you set the configuration title to. There was an option to skip the dynamic title added to watchtower, but it's not been released yet.

Would you mind creating a separate issue for `1` in [shoutrrr](https://github.com/containrrr/shoutrrr/issues/new/choose)?
